### ISSUE TITLE
Mongoose `find` with discriminator behavior changed from 5.x to 6.x

### ISSUE THREAD

**Issue by hasezoey** [Automattic/mongoose#10927]
**Do you want to request a *feature* or report a *bug*?**
bug / docs

**What is the current behavior?**
The behavior changed between mongoose 5.x and 6.x for a `.find` call with discriminators, and i could not see any related changes in the changelog or migration guide

i also think this may be related to the mongodb driver upgrade to 4.x

**If the current behavior is a bug, please provide the steps to reproduce.**

<details>
<summary>Script</summary>

For more details and comparison, please look at the reproduction branch

```ts
// NodeJS: 16.11.0
// MongoDB: 4.2-bionic (Docker)
import * as mongoose from 'mongoose'; // mongoose 6.0.12

const AnimalSchema = new mongoose.Schema(
  {
    patientNumber: {
      required: true,
      unique: true,
      type: String,
    },
  },
  { collection: 'animal' }
);

const DogSchema = AnimalSchema.clone();
DogSchema.add({ cageNumber: { type: String } });

const CatSchema = AnimalSchema.clone();
CatSchema.add({ nameTag: { type: String } });

const AnimalModel = mongoose.model('Animal', AnimalSchema);
const DogModel = AnimalModel.discriminator('Dog', DogSchema, 'Dog');
const CatModel = AnimalModel.discriminator('Cat', CatSchema, 'Cat');

(async () => {
  await mongoose.connect(`mongodb://localhost:27017/`, {
    dbName: 'verifyMASTER',
  });

  await mongoose.connection.db.dropDatabase();

  const doc1 = await AnimalModel.create({ patientNumber: 'animal-001' });
  const doc2 = await DogModel.create({ patientNumber: 'dog-002', cageNumber: 'dog-002' });
  const doc3 = await CatModel.create({ patientNumber: 'cat-003', nameTag: 'cat-003' });

  const results = await AnimalModel.find({ cageNumber: 'dog-xxx' });
  // on mongoose 5.x the result is: [] (expected)
  // on mongoose 6.x the result is: all 3 documents
  console.log('result', results);

  const found = await AnimalModel.find({ cageNumber: 0 }).exec();
  // on mongoose 5.x the result is: [] (expected)
  // on mongoose 6.x the result is: all 3 documents
  console.log('found', found);

  await mongoose.disconnect();
})();
```

</details>

reproduction repository / branch: https://github.com/typegoose/typegoose-testing/tree/mongooseDiscriminator6x

**What is the expected behavior?**

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
nodejs: 16.11.0
typescript: 4.4.4

old:
mongoose: 5.13.12

new:
mongoose: 6.0.12

**Comment by jamesweb1** [comment]
@hasezoey I should use mongoose with v5 before [#10490](https://github.com/Automattic/mongoose/issues/10933) got fixed. Is it correct?

**Comment by hasezoey** [comment]
> @hasezoey I should use mongoose with v5 before [#10490](https://github.com/Automattic/mongoose/issues/10933) got fixed. Is it correct?

no, you can use mongoose v6, it is already fixed, my issue (https://github.com/Automattic/mongoose/issues/10933) just says that the option *is not in the types yet*, so typescript will give you a error, but can be safely ignored

**Comment by hasezoey** [comment]
small update to my last message, the issue https://github.com/Automattic/mongoose/issues/10933 got fixed (which adds the option to the types), which will be included in mongoose 6.0.13
